# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: 6f19f43e0a0b
# Full Hash: 6f19f43e0a0bad72fa370191f4cc7f86f58566c2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 073302d26c5e -r 6f19f43e0a0b dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Thu Jun 11 06:37:38 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Thu Jun 11 06:37:40 2020 +0000
@@ -723,8 +723,6 @@
   bool mIsCanvasDirty = false;
   uvec2 mRequestedSize = {};
 
-  RefPtr<gfx::DataSourceSurface> mFrontBufferSnapshot;
-
  public:
   explicit ClientWebGLContext(bool webgl2);
 
@@ -997,7 +995,8 @@
   void Present(WebGLFramebufferJS*, layers::TextureType);
   Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebufferJS*,
                                                   layers::TextureType);
-  RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot() override;
+  RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot(
+      bool requireAlphaPremult = true) override;
 
  private:
   RefPtr<gfx::DataSourceSurface> BackBufferSnapshot();